crates/optimism/cli/src/commands/init_state/mod.rs

```diff
@@ -0,0 +1,99 @@
+//! Command that initializes the node from a genesis file.
+
+use clap::Parser;
+use reth_chainspec::ChainSpec;
+use reth_cli::chainspec::ChainSpecParser;
+use reth_cli_commands::common::{AccessRights, Environment, EnvironmentArgs};
+use reth_db_common::init::init_from_state_dump;
+use reth_node_builder::NodeTypesWithEngine;
+use reth_optimism_primitives::bedrock::BEDROCK_HEADER;
+use reth_provider::{
+    BlockNumReader, ChainSpecProvider, StaticFileProviderFactory, StaticFileWriter,
+};
+use std::{fs::File, io::BufReader, path::PathBuf};
+use tracing::info;
+
+mod bedrock;
+
+/// Initializes the database with the genesis block.
+#[derive(Debug, Parser)]
+pub struct InitStateCommand<C: ChainSpecParser> {
+    #[command(flatten)]
+    env: EnvironmentArgs<C>,
+
+    /// JSONL file with state dump.
+    ///
+    /// Must contain accounts in following format, additional account fields are ignored. Must
+    /// also contain { "root": \<state-root\> } as first line.
+    /// {
+    ///     "balance": "\<balance\>",
+    ///     "nonce": \<nonce\>,
+    ///     "code": "\<bytecode\>",
+    ///     "storage": {
+    ///         "\<key\>": "\<value\>",
+    ///         ..
+    ///     },
+    ///     "address": "\<address\>",
+    /// }
```

Comment:
```suggestion
    /// Must contain accounts in following format, additional account fields are ignored
    /// ```
    /// {
    ///     "balance": "\<balance\>",
    ///     "nonce": \<nonce\>,
    ///     "code": "\<bytecode\>",
    ///     "storage": {
    ///         "\<key\>": "\<value\>",
    ///         ..
    ///     },
    ///     "address": "\<address\>",
    /// }
    /// ```
    /// Must also contain `{ "root": \<state-root\> }` as first line.
```